hs_core/views/__init__.py

```diff
@@ -495,20 +497,24 @@ def update_metadata_element(request, shortkey, element_name, element_id, *args,
 @api_view(['GET'])
 def file_download_url_mapper(request, shortkey):
     """ maps the file URIs in resourcemap document to django_irods download view function"""
-
-    resource, _, _ = authorize(request, shortkey, needed_permission=ACTION_TO_AUTHORIZE.VIEW_RESOURCE)
-    istorage = resource.get_irods_storage()
-    irods_split = request.path.split('/')[2:-1]
-    irods_file_path = '/'.join(irods_split)
-    # [0:-1] excludes the last item on the list
-    listing = istorage.listdir('/'.join(irods_split[0:-1]))
-    if irods_split[-1] in listing[0]:
-        # it's a folder
-        file_download_url = istorage.url(os.path.join('zips', irods_file_path))
-        return HttpResponseRedirect(file_download_url)
-    else: 
-        file_download_url = istorage.url(irods_file_path)
-        return HttpResponseRedirect(file_download_url)
+    try: 
+        res, _, _ = authorize(request, shortkey,
+                              needed_permission=ACTION_TO_AUTHORIZE.VIEW_RESOURCE, 
+                              raises_exception=False)
+    except ObjectDoesNotExist: 
+        return HTTPResponse("resource not found", status=status.HTTP_404_NOT_FOUND) 
+    except PermissionDenied: 
+        return HTTPResponse("access not authorized", status=status.HTTP_401_UNAUTHORIZED) 
+
+    istorage = res.get_irods_storage()
+    logger.debug("request path is {}".format(request.path))
+    path_split = request.path.split('/')[2:]  # strip /resource/
+    public_file_path = '/'.join(path_split)
+    # logger.debug("public_file_path is {}".format(public_file_path))
+    istorage = res.get_irods_storage()
+    file_download_url = istorage.url(public_file_path)
+    logger.debug("redirect is {}".format(file_download_url))
```

Comment:
remove several logger debug statements above.

Reply:
Again, I would rather wait until I am sure this is working. No effect in production mode.

Reply:
I'm okay with debug loggers sticking around.

Reply:
@hyi I am in the process of scoping all of these inside `if __debug_:` contexts. Thus, they will only be run when in debug mode, and not in production.